// nodejs-colors/index.js
module.exports = {
    reset: () => {
        '\x1b[0m'
    },

    // Text Decoration
    bold: (text) => {
        console.log('\u001b[1m', text);
    },
    underline: (text) => {
        console.log('\u001b[4m', text);
    },

    // Colors
    black: (text) => {
        console.log('\u001b[30m', text);
    },
    red: (text) => {
        console.log('\x1b[31m', text);
    },
    green: (text) => {
        console.log('\u001b[32m', text);
    },
    yellow: (text) => {
        console.log('\u001b[33m', text);
    },
    blue: (text) => {
        console.log('\u001b[34m', text);
    },
    magenta: (text) => {
        console.log('\x1b[35m', text);
    },
    cyan: (text) => {
        console.log('\u001b[36m', text);
    },
    white: (text) => {
        console.log('\u001b[37m', text);
    },

    // Backgrounds
    bgBlack: (text) => {
        console.log('\u001b[40m', text);
    },
    bgRed: (text) => {
        console.log('\u001b[41m', text);
    },
    bgGreen: (text) => {
        console.log('\u001b[42m', text);
    },
    bgYellow: (text) => {
        console.log('\u001b[43m', text);
    },
    bgBlue: (text) => {
        console.log('\u001b[44m', text);
    },
    bgMagenta: (text) => {
        console.log('\u001b[45m', text);
    },
    bgCyan: (text) => {
        console.log('\u001b[46m', text);
    },
    bgWhite: (text) => {
        console.log('\u001b[47m', text);
    },
};
